src/screens/Ticketing/Ticket/ValidityLine.tsx

```diff
@@ -57,4 +57,104 @@ const ValidityLine: React.FC<{
   }
 };
 
+/**
+ * Starts an animation for the vertical line offset which loops continuously.
+ * The offset is animated from 0 to 1, which must be interpolated to the actual
+ * pixel offset value in the vertical line component.
+ * @returns the current animated offset value
+ */
+const useAnimatedVerticalLineOffset = () => {
+  const animatedOffset = useRef(new Animated.Value(0)).current;
+  useEffect(() => {
+    Animated.loop(
+      Animated.timing(animatedOffset, {
+        toValue: 1,
+        duration: 1000,
+        useNativeDriver: true,
+        easing: Easing.linear,
+      }),
+    ).start();
+  }, []);
+  return animatedOffset;
+};
+
+/**
+ * Calculates the validity percent based on time left on ticket. This value is
+ * used for determining how wide the animated part of the progress bar should
+ * be.
+ */
+const getValidityPercent = (
+  nowSeconds: number,
+  validFrom: number,
+  validTo: number,
+) => {
+  const durationSeconds = validTo - validFrom;
+  const timeLeftSeconds = validTo - nowSeconds;
+  return Math.ceil((timeLeftSeconds / durationSeconds) * 100);
+};
+
+/**
+ Get the number of vertical lines necessary to fill the progress bar. For
+ simplicity's sake it is calculated by using the window viewport width instead
+ of actual progress bar width. I found using the onLayout listener to calculate
+ the actual progress bar width was not worth the effort as some overflowing
+ hidden vertical lines doesn't hurt anybody.
+ */
+const getNumberOfVerticalLines = () =>
+  Math.ceil(Dimensions.get('window').width / SPACE_BETWEEN_VERTICAL_LINES);
+
+const VerticalLine = ({
+  offset,
+  index,
+}: {
+  offset: Animated.Value;
+  index: number;
+}) => (
+  <AnimatedLinearGradient
+    style={[
+      useStyles().verticalLine,
+      {
+        left: index * SPACE_BETWEEN_VERTICAL_LINES - 10,
+        transform: [
+          {
+            translateX: offset.interpolate({
+              inputRange: [0, 1],
+              outputRange: [SPACE_BETWEEN_VERTICAL_LINES, 0],
+            }),
+          },
+        ],
+      },
+    ]}
+    useAngle={true}
+    angle={120}
+    locations={[0.25, 0.25, 0.75, 0.75]}
+    colors={[
+      'transparent',
+      colors.primary.gray_500,
+      colors.primary.gray_500,
+      'transparent',
+    ]}
+    pointerEvents={'none'}
+  />
+);
+
+const useStyles = StyleSheet.createThemeHook((theme) => ({
+  container: {
+    marginVertical: theme.spacings.medium,
+    marginHorizontal: -theme.spacings.medium,
+    flexDirection: 'row',
+  },
+  progressBar: {
+    height: 8,
+    backgroundColor: colors.primary.green_500,
```

Comment:
Changes to theming in app code will be making use of themed color identifiers, as they are defined in Figma. I think green mostly just stay the same green, but many colors will change between light and dark (and potential future themes). Not something you can solve now, just a note that we should remember to change these color references later (I imagine that we'll need to refactor many places due to the new design system, so it'll probably be solved then).

Reply:
Yeah, was working on this when I asked about the colors on Slack. The colors in dark mode is ok-ish, but probably not exactly how @pettersuul envisioned.